pkg/kgo/broker.go

```diff
@@ -793,7 +794,8 @@ start:
 	// api versions does *not* use flexible response headers; see comment in promisedResp
 	rawResp, err := cxn.readResponse(nil, req.Key(), req.GetVersion(), corrID, false, rt, bytesWritten, writeWait, timeToWrite, readEnqueue)
 	if err != nil {
-		if strings.HasSuffix(err.Error(), "connection reset by peer") {
+		var errno syscall.Errno
+		if errors.As(err, &errno) && isConnReset(errno) {
```

Comment:
In my humble (unrequested) opinion, maybe this is more elegant if the check function is defined as:
```go
func isConnResetErr(err error) bool
```

That way `syscall` does not need to pollute this file and there is no need to instantiate a variable here, but on the check function itself. Good fix anyway!

Reply:
I think that's a good point, however on the other hand, doing it here once avoids needing to do the same errors.As in both check files. I think I'll merge with this, but both are great options :)